base/ltoutput.dtx

```diff
@@ -2350,9 +2402,18 @@
 %    Then the footnotes are appended:
 %    \begin{macrocode}
      \@outputbox@append{%
+%    \end{macrocode}
+%    First we call a socket which may or may not back up by the depth
+%    of the \cs{@outputbox} so that we start adding from the baseline
```

Comment:
"start adding" what? 

Maybe better:
. . . so that the added skip starts at the baseline of the last line.